src/backend/distributed/master/master_truncate.c

```diff
@@ -139,3 +142,37 @@ TruncateTaskList(Oid relationId)
 
 	return taskList;
 }
+
+
+/*
+ * truncate_local_table truncates the local records of a distributed table.
+ *
+ * The main advantage of this function is to truncate all local records after creating a
+ * distributed table, and prevent constraints from failing due to outdated local records.
+ */
+Datum
+truncate_local_table(PG_FUNCTION_ARGS)
```

Comment:
This naming seems a bit confusing to me. If we had a function to truncate a local table, it would be this. But this function is about local shards of a distributed table, do I understand that correctly?

Does this PR aim to solve the following:
- We have some local table
- We insert some data to the local table
- We distribute the table
- The local data is still there and this can cause constraint problems even though this data is not used?

So if that is the case, why don't we call this function when we are distributing the table in the first place so that users don't have to call this function explicitly.

Reply:
This function is not about truncating local shards. It just removes all those records that were copied to the shards, and now are not accessible.

Quoting [Marco's comment](https://github.com/citusdata/citus/issues/1770#issuecomment-446205988)

> +1 on fixing this (by truncating the local table after a create_distributed_table)
> 
> While some users do want to go back to having a local table after distributing a table, they often already accepted writes on the distributed table so having an old copy of the data is not useful. It may be worth adding an undistribute_table function instead.

i believe we can safely truncate local records after we introduce a `undistribute_table` function

Reply:
Do you have suggestion for a better name? I am not so happy with the current one either, but I could not find a better alternative

Reply:
maybe:
truncate_distributed_table_old_data
truncate_distributed_table_unused_data (as it is unused)

Reply:
I think it is important to explicitly emphasize that this is for some unused, old data. Do my suggestions make sense?

I also suggest to update the description as:

```
DESCRIPTION: Introduces XXX UDF to truncate unused/old local records
```

Reply:
I updated the description and changed the name of the UDF to `truncate_local_data_after_distributing_table()`